pipeline/transformer/cifs.py

```diff
@@ -0,0 +1,284 @@
+                    # FIXME the BW publication does not contain a LineString. As this is required by cifs, we add a minimal offset as workaround
+                    geometry = '{} {} {} {}'.format(lat, lon, lat, lon + 0.00001)
+                    geojsonGeometry = {'type': 'Point', 'coordinates': [lon, lat]}
+                else:
+                    geometry = polyline.text
+                    geojsonGeometry = {
+                        'type': 'LineString',
+                        'coordinates': self._pairwise([float(i) for i in geometry.split()]),
+                    }
+
+                (starttime, endtime) = self._get_start_end_time(situationRecord)
+                location = {
+                    'polyline': geometry,
+                    'street': self._road_name(situationRecord),
+                    'direction': self._detect_direction(situation, situationRecord),
+                }
+                closure = {
+                    'id': situationRecord.get('id'),
+                    'type': self._incident_type(situationRecord),
+                    'subtype': self._incident_subtype(situationRecord),
+                    'starttime': starttime,
+                    'endtime': endtime,
+                    'description': self._roadworks_name(overallSituation),
+                    'reference': self.reference,
+                }
+
+                if 'geojson' == format:
+                    closure['street'] = location.get('street')
+                    closure['direction'] = location.get('direction')
+                    feature = {'type': 'Feature', 'geometry': geojsonGeometry, 'properties': closure}
+                    features.append(feature)
+                else:
+                    closure['location'] = location
+                    closures.append(closure)
+
+        if 'geojson' == format:
+            geojson = {'type': 'FeatureCollection', 'features': features}
+            json_result = geojson
+        else:
+            incidents = {'incidents': closures, 'timestamp': datetime.datetime.now().isoformat()}
+            json_result = incidents
+
+        return json_result
+
+    def transform(self, datex2file: str, format: str = 'cifs') -> dict:
+        '''
+        Transforms situation records into cifs-roadworks, like e.g.:
+        [{
+          "id": "101",
+          "type": "ROAD_CLOSED",
+          "subtype": "ROAD_CLOSED_CONSTRUCTION",
+          "polyline": "51.510090 -0.006902 51.509142 -0.006564 51.506291 -0.003640 51.503796 0.001051 51.499218 0.001687 51.497365 0.002020",
+          "street": "NW 12th St",
+          "starttime": "2016-04-07T09:00:00+01:00",
+          "endtime": "2016-04-07T23:00:00+01:00",
+          "description": "Closure on I-95 NB due to construction",
+          "direction": "BOTH_DIRECTIONS"
+        },
+        ...
+        ]
+        '''
+
+        return self.transform_datex2(self._parse(datex2file), format)
+
+
+if __name__ == '__main__':
```

Comment:
This DATEX2 -> CIFS/GeoJSON converter has applications far beyond MobiData BW's IPL. To attract third-party contributions – which might also increase its long-term maintainability –, we might want to move it into a separate CLI/lib/repo.

Reply:
Had the same thought: why not re-use the generic XML2JSON + validation we use in ParkAPI? Would require an own project. But I guess this will be something beyond this MR, and so I don't get too much into details of the code.